app/assets/javascripts/discourse/app/lib/plugin-api.js

```diff
@@ -1926,6 +1927,37 @@ class PluginApi {
   registerUserMenuTab(func) {
     registerUserMenuTab(func);
   }
+
+  /**
+   * EXPERIMENTAL. Do not use.
+   * Apply transformation using a callback on a list of model instances of a
+   * specific type. Currently, this API only works on lists rendered in the
+   * user menu such as notifications, bookmarks and topics (i.e. messages), but
+   * it may be extended to other lists in other parts of the app.
+   *
+   * You can pass an `async` callback to this API and it'll be `await`ed and
+   * block rendering until the callback finishes executing.
+   *
+   * ```
+   * api.registerModelTransformer("topic", (topics) => {
```

Comment:
```suggestion
   * api.registerModelTransformer("topic", async (topics) => {
```

I think this function needs to be `async`